client/src/client.rs

```diff
@@ -715,8 +728,12 @@ impl Client {
             .send()?;
 
             if response.status() == StatusCode::OK {
-                let pending_transactions =
-                    VersionedPendingTransactions::decode_versioned(response.body())?;
+                let mut res = VersionedPendingTransactions::decode_all_versioned(response.body());
+                if res.is_err() {
+                    warn!("Can't decode pending transactions using all bytes");
```

Comment:
```suggestion
                    warn!("Failed to decode pending transactions, not all bytes were consumed");
```
Which should be the only reason for failure.